src/travel-details-screens/DepartureDetailsScreenComponent.tsx

```diff
@@ -161,6 +166,43 @@ export const DepartureDetailsScreenComponent = ({
     (estimatedCall) => estimatedCall.quay?.id === activeItem.fromQuayId,
   );
 
+  const {onMarkFavourite, getExistingFavorite} = useOnMarkFavouriteDepartures(
+    fromQuay?.quay,
+    fromQuay?.quay.stopPlace,
+    false,
+  );
+
+  const favouriteDepartureLine = fromQuay?.quay.id
+    ? {
+        ...serviceJourney?.line,
+        lineNumber: publicCode,
+        destinationDisplay: fromQuay?.destinationDisplay,
+        id: fromQuay.quay.id,
+      }
+    : undefined;
+  const existingFavorite = getExistingFavorite(favouriteDepartureLine);
+  const favouriteIcon = getFavoriteIcon(existingFavorite);
+  const accessibilityLabel = () => {
+    if (!existingFavorite) {
+      return '';
+    } else if (existingFavorite.destinationDisplay) {
+      return t(DeparturesTexts.favorites.favoriteButton.oneVariation);
+    } else {
+      return t(DeparturesTexts.favorites.favoriteButton.allVariations);
+    }
+  };
+
+  const onCloseFocusRef = useRef<RefObject<any>>(null);
+  const onPressFavorite = () => {
+    if (favouriteDepartureLine) {
+      onMarkFavourite(
+        favouriteDepartureLine,
+        existingFavorite,
+        onCloseFocusRef,
+      );
+    }
+  };
+
```

Comment:
Yes, that code is due to the nature of the base code, everything is everywhere and code duplication because types are not reusable and are quite different even tif hey have the same data, it could be moved out yes but refactor the `FavoriteDepartureButton` to make it seamless might introduce more complexity to the actual issue. So move the code out to a `useOnMarkFavouriteDepartures` could work.